src/Altinn.App.Core/Features/Validation/ExpressionValidator.cs

```diff
@@ -0,0 +1,276 @@
+        {
+            var rawValidationConfig = appResourceService.GetValidationConfiguration(dataType);
+            if (rawValidationConfig == null)
+            {
+                // No validation configuration exists for this data type
+                return new List<ValidationIssue>();
+            }
+
+            var validationConfig = JsonDocument.Parse(rawValidationConfig).RootElement;
+            return Validate(validationConfig, dataModel, evaluatorState, logger);
+        }
+
+        /// <inheritdoc />
+        public static IEnumerable<ValidationIssue> Validate(JsonElement validationConfig, IDataModelAccessor dataModel, LayoutEvaluatorState evaluatorState, ILogger logger)
+        {
+            var validationIssues = new List<ValidationIssue>();
+            var expressionValidations = ParseExpressionValidationConfig(validationConfig, logger);
+            foreach (var validationObject in expressionValidations)
+            {
+                var baseField = validationObject.Key;
+                var resolvedFields = dataModel.GetResolvedKeys(baseField);
+                var validations = validationObject.Value;
+                foreach (var resolvedField in resolvedFields)
+                {
+                    var positionalArguments = new[] { resolvedField };
+                    foreach (var validation in validations)
+                    {
+                        try
+                        {
+                            if (validation.Condition == null)
+                            {
+                                continue;
+                            }
+
+                            var isInvalid = ExpressionEvaluator.EvaluateExpression(evaluatorState, validation.Condition, null, positionalArguments);
+                            if (isInvalid is not bool)
+                            {
+                                throw new ArgumentException($"Validation condition for {resolvedField} did not evaluate to a boolean");
+                            }
+                            if ((bool)isInvalid)
+                            {
+                                var validationIssue = new ValidationIssue
+                                {
+                                    Field = resolvedField,
+                                    Severity = validation.Severity ?? ValidationIssueSeverity.Error,
+                                    CustomTextKey = validation.Message,
+                                    Code = validation.Message,
+                                    Source = ValidationIssueSources.Expression,
+                                };
+                                validationIssues.Add(validationIssue);
+                            }
+                        }
+                        catch
+                        {
+                            logger.LogError($"Error while evaluating expression validation for {resolvedField}");
+                            throw;
```

Comment:
Æsj, jeg burde sett dette før.
Det er lurt å sende med exception i loggen, og også gjøre meldingen parametrisk så man kan lage bedre queries i application insights.

```suggestion
                        catch(Exception e)
                        {
                            logger.LogError(e, "Error while evaluating expression validation for {resolvedField}", resolvedField);
                            throw;
```

Reply:
:+1: nice catch, I missed that one 🙈 fix in new PR: https://github.com/Altinn/app-lib-dotnet/pull/319